cml-tdengine/src/core/inference.rs

```diff
@@ -298,77 +295,117 @@ mod tests {
             b"20",
         )?;
         let inference_fn = |vec_data: &mut Vec<NewSample<Value>>,
                             batch: &str,
                             task_time: i64|
          -> Vec<NewSample<Value>> {
             let mut result: Vec<NewSample<Value>> = Vec::new();
-            let working_dir = "/tmp/inference_dir/".to_string();
-            let model_inference =
-                fs::read_to_string(working_dir + batch + &task_time.to_string() + ".txt")
-                    .unwrap()
-                    .parse::<f32>()
-                    .unwrap();
+            let working_dir = env::temp_dir().join("inference_dir/");
+            let model_inference = fs::read_to_string(
+                working_dir.join(batch.to_string() + &task_time.to_string() + ".txt"),
+            )
+            .unwrap()
+            .parse::<f32>()
+            .unwrap();
             for inference_data in vec_data.iter() {
                 // inference
-                let inference_result = fs::read_to_string(inference_data.data_path())
-                    .unwrap()
-                    .parse::<f32>()
-                    .unwrap()
+                let inference_result = fs::read_to_string(
+                    inference_data
+                        .optional_fields()
+                        .as_ref()
+                        .unwrap()
+                        .first()
+                        .unwrap()
+                        .to_string()
+                        .unwrap(),
+                )
+                .unwrap()
+                .parse::<f32>()
+                .unwrap()
                     + model_inference;
                 let output = if inference_result > 25.0 {
                     Some(Value::Float(inference_result))
                 } else {
-                    None
+                    Some(Value::Null(Ty::Float))
                 };
                 result.push(
-                    NewSampleBuilder::default()
-                        .data_path(inference_data.data_path().to_path_buf())
-                        .output(output)
-                        .build()
-                        .unwrap(),
+                    NewSample::builder()
+                        .optional_fields(
+                            inference_data.optional_fields().as_ref().unwrap().to_vec(),
+                        )
+                        .output(output.unwrap())
+                        .build(),
                 );
             }
             result
         };
-
-        tokio::spawn({
+        let batch_state = Arc::new((Mutex::new(HashSet::new()), Condvar::new()));
+        let cml_2 = cml.clone();
+        let batch_state_2 = batch_state.clone();
+        let pool_2 = pool.clone();
+        let cml_3 = cml.clone();
+        let batch_meta_2_dup = batch_meta_2.clone();
+        let batch_state_3 = batch_state.clone();
+        let pool_3: deadpool::managed::Pool<Manager<TaosBuilder>> = pool.clone();
```

Comment:
It would be better to use `Arc<T>`. Although we may not be concerned about the minor performance difference in tests, this case could lead to user misunderstandings.